src/QueueFactory.php

```diff
@@ -1,192 +0,0 @@
-        }
-
-        if ($this->enableRuntimeChannelDefinition === false) {
-            throw new ChannelNotConfiguredException($channel);
-        }
-
-        /** @psalm-suppress PossiblyNullReference */
-        return $this->queue->withChannelName($channel)->withAdapter($this->defaultAdapter->withChannel($channel));
-    }
-
-    private function checkDefinitionType(string $channel, mixed $definition): void
-    {
-        if (
-            !$definition instanceof AdapterInterface
-            && !is_array($definition)
-            && !is_callable($definition)
-            && !is_string($definition)
-        ) {
-            throw new ChannelIncorrectlyConfigured($channel, $definition);
-        }
-    }
-
-    public function createFromDefinition(
-        string $channel,
-        AdapterInterface|callable|array|string $definition
-    ): AdapterInterface {
-        if ($definition instanceof AdapterInterface) {
-            return $definition;
-        }
-
-        if (is_string($definition)) {
-            return $this->getFromContainer($channel, $definition);
-        }
-
-        return $this->tryGetFromCallable($channel, $definition)
-            ?? $this->tryGetFromArrayDefinition($channel, $definition)
-            ?? throw new ChannelIncorrectlyConfigured($channel, $definition);
-    }
-
-    private function getFromContainer(string $channel, string $definition): AdapterInterface
-    {
-        if (class_exists($definition)) {
-            if (is_subclass_of($definition, AdapterInterface::class)) {
-                /** @var AdapterInterface */
-                return $this->container->get($definition);
-            }
-        } elseif ($this->container->has($definition)) {
-            $middleware = $this->container->get($definition);
-            if ($middleware instanceof AdapterInterface) {
-                return $middleware;
-            }
-        }
-
-        throw new ChannelIncorrectlyConfigured($channel, $definition);
-    }
-
-    private function tryGetFromCallable(
-        string $channel,
-        callable|AdapterInterface|array|string $definition
-    ): ?AdapterInterface {
-        $callable = null;
-
-        if ($definition instanceof Closure) {
-            $callable = $definition;
-        }
-
-        if (
-            is_array($definition)
-            && array_keys($definition) === [0, 1]
-        ) {
-            try {
-                $callable = $this->callableFactory->create($definition);
-            } catch (InvalidCallableConfigurationException $exception) {
-                throw new ChannelIncorrectlyConfigured($channel, $definition, previous: $exception);
-            }
-        }
-
-        if ($callable !== null) {
-            $adapter = $this->injector->invoke($callable);
-
-            if (!$adapter instanceof AdapterInterface) {
-                throw new ChannelIncorrectlyConfigured($channel, $definition);
-            }
-        }
-
-        return null;
-    }
```

Comment:
Looks like this definition type is not implemented in providers. Am I right?

Reply:
`QueueFactoryQueueProvider` and `AdapterFactoryQueueProvider` supported all definitions that Yii Factory support, including callable definitions.

Reply:
Does it support `[ClassName::class, 'method']` format? I'm not sure about that.

Reply:
See https://github.com/yiisoft/definitions?tab=readme-ov-file#callabledefinition